### ISSUE TITLE
Task prerequisites duplication

### ISSUE THREAD

**Issue by ColemanTom** [cylc/cylc-flow#5465]
### Description

This was something I noticed in Cylc7, and just checked if it was still the case in Cylc 8.1.0. It is.

When you do certain graph constructs, task prerequisites can be duplicated. It's not optimal to look at, but I'm going to guess, its not optimal for Cylc internals either. Are the prerequisites stored as a list instead of a set? The off-topic discussion in [this threaad](https://cylc.discourse.group/t/cylc-reinstall-auto-yes/610/18) made me think to test this item.

### Reproducible Example

Here is a contrived example

```
[scheduler]
   allow implicit tasks=True

[task parameters]
  run = 1..2

[scheduling]
  initial cycle point = 20200101
  final cycle point = 20200103
  [[ graph]]
    P1D = """
      a<run> => c
      b[-P1D] => b
      b => c
    """
    PT12H = """
      a<run> => c
      b[-P1D] => b
      b => c
    """

[runtime]
  [[root]]
    script = echo hello
```

Launching that and doing `cylc show workflow//20200102T0000Z/c`, we get this list of prerequisites.

```
prerequisites: ('-': not satisfied)
  - 20200102T0000Z/a_run2 succeeded
  - 20200102T0000Z/b succeeded
  - 20200102T0000Z/a_run1 succeeded
  - 20200102T0000Z/a_run2 succeeded
  - 20200102T0000Z/b succeeded
  - 20200102T0000Z/a_run1 succeeded
```

So there are 6 listed, but really it is only 3, its just been duplicated.

Similarly, if I do `cylc show workflow//20200102T0000Z/b` we see two prerequisites instead of one.

```
prerequisites: ('-': not satisfied)
  - 20200101T0000Z/b succeeded
  - 20200101T0000Z/b succeeded
```

### Expected Behaviour

As described above, no duplication in task prerequisites.

**Comment by hjoliver** [comment]
Thanks for that @ColemanTom 

I think what's going on is that prerequisites are recurrence-specific (i.e. `P1D` and `PT12H` in this case) and we're not eliminating duplicates when multiple recurrences generate the same concrete prerequisites at a particular cycle point.

I don't think this would cause problems internally, other than a small inefficiency, but definitely confusing for users!

**Comment by ColemanTom** [comment]
Thanks. So it is really a visualisation issue. Good to know, I was always curious but never had bothered to ask.

One extra request if this were to be fixed, perhaps the outputs could be sorted using your natural sort methods? In my example you can see the prerequisites aren't listed in a great order

**Comment by ColemanTom** [comment]
I think this can be closed? The sorting could be a separate issue if desired.

**Comment by hjoliver** [comment]
(Yes, thanks for the reminder - we sometimes forget that issues don't get closed automatically when merging a PR to a non-master branch)
